### PR TITLE
React2 Reiteration

### PR DESCRIPTION
**Pull request by chezzoba**
Made changes to React 2.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -1,69 +1,58 @@
...
+ - Why we learn React
+ - How to write React components in **JSX**
+ - How to use **props** and **state** in components
+ - How to react to events using **hooks**
 
-- http://es6-features.org/#Constants
-- http://es6-features.org/#ObjectMatchingShorthandNotation
-- http://es6-features.org/#RestParameter
-- http://es6-features.org/#SpreadOperator
+### [React 2](/react2/readme.md) (2 weeks)
```
Comment (adamblanchard) [inline readme.md:36]: "2 weeks" here conflicts a little with the link it goes to.

I suggest we change it here to 3 weeks and include the meal sharing bit in this section too, like:

```
### [React 2](/react2/readme.md) (3 weeks)
...
In the third week, you'll get to implement the UI for your Meal Sharing app...
```

File: react1/readme.md [hunk 2]
```diff
@@ -2,10 +2,8 @@
...
-| 2.   | **Building components** Component tree <br> <li> From sketch of todolist to component tree <br> Stateful logic - Having logic around state <br> List keys <br> Life cycle <br> Lifting state                                                                                                                                                                                                                  | [Preparation](week2/preparation.md) | [Lesson plan](week2/lesson-plan.md) | [Homework](week2/homework.md) |
-| 3.   | **Advanced React** <br> Forms and events <br> Using data fetching <br> Children <br> Prop types                                                                                                                                                                                                                                                                                                               | [Preparation](week3/preparation.md) | [Lesson plan](week3/lesson-plan.md) | [Homework](week3/homework.md) |
+| Week | Topic | Preparation | Lesson plan | Homework |
```
Comment (adamblanchard) [inline react1/readme.md:5]: For consistency with the other modules, can the homework links go to the readme instead of directly to the repo?

**Review by adamblanchard — CHANGES_REQUESTED** [review]

**Review by markitosha — COMMENTED** [review]
This is an amazing piece of work you've done here!
I have a couple of thoughts regarding changes for the program, and would like to discuss them here. Disclaimer: if I sound too opinionated or harsh somewhere, perhaps it's a language barrier and I'm sorry 😅 

---

While I agree that React 2 should focus on Next.js, I’m not sure if introducing it in React 1 is the best approach.

In my opinion, React 1 should stick to plain React (using Vite or something similar) because it allows students to learn the fundamentals without the added complexity of a framework. Introducing Next.js too early could lead to framework-specific errors that are difficult to address in the first few weeks. (I still encounter them even on small projects, and I don’t think we want to teach people to ignore warnings 😄 ). Plain React, as a library, is simple and effective for mastering basics. Also we might focus only on core basics in the first week and we won't have to introduce routing that quickly. 

From a practical standpoint, many companies still use plain React or their own custom SSR solutions. By not pushing students towards a framework too early, we could help make them more adaptable to various approaches (although I admit this point is somewhat theoretical 😄).

As the rest of the program,
- I found missing a ["server actions"](https://nextjs.org/docs/app/building-your-application/data-fetching/server-actions-and-mutations) topic for Next.js (if we want them to use SSR -- this is the way)
- That would be nice to mention "custom hooks" (very small topic, but can decrease a code complexity a lot simply by code splitting)
- This one I'm not sure about: other concepts, such as easy optimizations (useMemo, useCallback, React.memo) and authentication, can also be introduced once students have a solid foundation. But maybe they are too advanced (but usually wanted for the jobs) 

Again, I'm here almost "first day" and don't know students very well, so everything here is open for discussion!

File: react1/week3/lesson-plan.md [hunk 3]
```diff
@@ -1,166 +1,91 @@
...
-      <li key={randomCat.id}>
-        <FancyBorder>
-          <img src={randomCat.url} alt="Random cat" />
-        </FancyBorder>
-      </li>
-    );
-  });
+### Rendering multiple components (10 min)
+* Rendering components using `map()`
+* `key` property
```
[the remark this answers is not in the document]
Comment (chezzoba) [reply react1/week3/lesson-plan.md:16]: Yes, thanks! Removed that exercise there.

File: react1/week2/lesson-plan.md [hunk 4]
```diff
@@ -1,114 +1,166 @@
...
+### Setting attributes (5 min)
+* HTML attributes
+* className
+
+### Importing / exporting .jsx (5 min)
+* defauls exports and named exports
+
+### Importing .css (5 minutes)
+* Global styles vs. component-specific styles
+* CSS modules in Next.js
```
[the remark this answers is not in the document]
Comment (chezzoba) [reply react1/week2/lesson-plan.md]: You're right. Removed it here.

**Review by adamblanchard — APPROVED** [review]

**Pull request merged**